### PR TITLE
Improve handling of ambiguous refnames

### PR DESCRIPTION
**Pull request by kyleam**
Please see commit messages.

Re: #3098

### CODE REVIEW

File: lisp/magit-diff.el [hunk 1]
```diff
@@ -1339,7 +1339,11 @@ commit or stash at point, then prompt for a commit."
           (user-error "No commit on this line"))))
      (t
       (magit-section-case
-        ((commit branch)
+        (branch
+         (setq rev (magit-ref-maybe-qualify (magit-section-value it))
+               cmd 'magit-show-commit
+               buf (magit-mode-get-buffer 'magit-revision-mode)))
```
Comment (tarsius) [inline lisp/magit-diff.el]: Please use three `setq`s. Stefan says that the ability to set multiple variables is only supported for backward compatibility.

Comment (kyleam) [inline lisp/magit-diff.el]: That multi-symbol style is used throughout
`magit-diff-show-or-scroll`.  It'd be better to change all the
single-symbol setq's in a separate step, no?

Comment (kyleam) [inline lisp/magit-diff.el]: > It'd be better to change all the single-symbol setq's in a separate step

The branch [km/multi-setq] should take care of all the multi-symbol
setq's in Magit, if that's the direction you want to go.

[km/multi-setq]: https://github.com/magit/magit/compare/km/multi-setq

Comment (kyleam) [inline lisp/magit-diff.el]: Done.  Thanks for the review.

**Pull request merged**